app/models/manageiq/providers/ibm_cloud/object_storage/storage_manager.rb

```diff
@@ -0,0 +1,211 @@
+class ManageIQ::Providers::IbmCloud::ObjectStorage::StorageManager < ManageIQ::Providers::StorageManager
+  require_nested :Refresher
+  require_nested :RefreshWorker
+
+  supports :ems_storage_new
+  include ManageIQ::Providers::StorageManager::ObjectMixin
+
+  def self.params_for_create
+    @params_for_create ||= {
+      :fields => [
+        {
+          :component => 'sub-form',
+          :id        => 'endpoints-subform',
+          :name      => 'endpoints-subform',
+          :title     => _("Endpoints"),
+          :fields    => [
+            :component => 'tabs',
+            :name      => 'tabs',
+            :fields    => [
+              {
+                :component => 'tab-item',
+                :id        => 'default-tab',
+                :name      => 'default-tab',
+                :title     => _('Default'),
+                :fields    => [
+                  {
+                    :component              => 'validate-provider-credentials',
+                    :id                     => 'endpoints.default.valid',
+                    :name                   => 'endpoints.default.valid',
+                    :skipSubmit             => true,
+                    :isRequired             => true,
+                    :validationDependencies => %w[type zone_id uid_ems],
+                    :fields                 => [
+                      {
+                        :component  => "text-field",
+                        :name       => "provider_region",
+                        :id         => "provider_region",
+                        :label      => _("Region"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
```

Comment:
Best for this to be a dropdown, is this the same list of regions as VPC?  See https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/vpc/manager_mixin.rb#L42 and https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager.rb#L65-L75

They are loaded from a yaml file: https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/db/fixtures/ibm_cloud_vpc_regions.yml

Reply:
The list is a bit bigger than that for COS. There are regional, global and some other types of locations. See here: https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-endpoints